localstack/aws/connect.py

```diff
@@ -466,14 +465,19 @@ def get_client(
             if re.match(r"https?://localhost(:[0-9]+)?", endpoint_url):
                 endpoint_url = endpoint_url.replace("://localhost", f"://{get_s3_hostname()}")
 
+        # Prevent `PartialCredentialsError` when only access key ID is provided
+        # The value of secret access key is insignificant and can be set to anything
+        if aws_access_key_id:
+            aws_secret_access_key = aws_secret_access_key or TEST_AWS_SECRET_ACCESS_KEY
+
         return self._get_client(
             service_name=service_name,
             region_name=region_name or config.region_name or self._get_region(),
             use_ssl=self._use_ssl,
             verify=self._verify,
             endpoint_url=endpoint_url,
-            aws_access_key_id=aws_access_key_id or TEST_AWS_ACCESS_KEY_ID,
```

Comment:
We don't want to use the test credentials in non-test scenarios. This is part of the effort to make accounts explicit and avoid silent fallbacks, making cross-account bugs tiresome to debug.

The value of secret access key is currently ignored by LS so that is excused.